pkg/minishift/config/instance_dirs.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"github.com/minishift/minishift/pkg/minikube/constants"
+)
+
+type MinishiftDirs struct {
+	Home       string
+	Config     string
+	Machines   string
+	Certs      string
+	Cache      string
+	IsoCache   string
+	OcCache    string
+	ImageCache string
+	Addons     string
+	Logs       string
+	Tmp        string
+}
+
+var InstanceDirs *MinishiftDirs
+
+//constructor for MinsihiftDirs
+func NewMinishiftDirs(profileName string) *MinishiftDirs {
+
+	dirs := &MinishiftDirs{}
+	dirs.Home = constants.Minipath
+	dirs.Certs = constants.MakeMiniPath("certs")
```

Comment:
using MakeMinePath makes the structure unclear at first glance.

Reply:
I do not think so. Thats pretty much straight forward if you check constants.go.